cli/azd/internal/appdetect/java.go

```diff
@@ -248,6 +257,31 @@ func detectDependencies(mavenProject *mavenProject, project *Project) (*Project,
 	return project, nil
 }
 
+func detectSpringBootVersion(currentRoot *mavenProject, mavenProject *mavenProject) string {
+	if currentRoot != nil {
+		if currentRoot.Parent.ArtifactId == "spring-boot-starter-parent" {
+			return currentRoot.Parent.Version
+		} else {
+			for _, dep := range currentRoot.DependencyManagement.Dependencies {
+				if dep.ArtifactId == "spring-boot-dependencies" {
+					return dep.Version
+				}
+			}
+		}
+	} else if mavenProject != nil {
+		if mavenProject.Parent.ArtifactId == "spring-boot-starter-parent" {
+			return mavenProject.Parent.Version
+		} else {
+			for _, dep := range mavenProject.DependencyManagement.Dependencies {
+				if dep.ArtifactId == "spring-boot-dependencies" {
+					return dep.Version
+				}
+			}
+		}
```

Comment:
Could you please create a function to avoid duplicated codes?

Reply:
Good idea.